client/app/(auth)/createProfile.tsx

```diff
@@ -1,19 +1,430 @@
+    source: require("../../assets/images/placeholder-avatar.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-1.png",
+    source: require("../../assets/images/profile/profile-1.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-2.png",
+    source: require("../../assets/images/profile/profile-2.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-3.png",
+    source: require("../../assets/images/profile/profile-3.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-4.png",
+    source: require("../../assets/images/profile/profile-4.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-5.png",
+    source: require("../../assets/images/profile/profile-5.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-6.png",
+    source: require("../../assets/images/profile/profile-6.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-7.png",
+    source: require("../../assets/images/profile/profile-7.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-8.png",
+    source: require("../../assets/images/profile/profile-8.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-9.png",
+    source: require("../../assets/images/profile/profile-9.png"),
+  },
+];
+
+const schema = yup.object({
+  avatarUrl: yup
+    .string()
+    .required("Avatar is required")
+    .test(
+      "not-default-avatar",
+      "Avatar is required",
+      (value) => value !== avatarOptions[0].url,
+    ),
+  dateOfBirth: yup.date().required("Date of Birth is required"),
+  username: yup.string().required("Username is required"),
+  name: yup.string().required("Name is required"),
+});
 
 const CreateProfile = () => {
-  const user = useAppSelector((state) => state.user);
+  const auth = getAuth();
+  const currentUser = auth.currentUser;
+
+  const dispatch = useAppDispatch();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedAvatarOption, setSelectedAvatarOption] =
+    useState<AvatarOption>(avatarOptions[0]);
+  const [showBottomDrawer, setShowBottomDrawer] = useState(false);
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [selectedDate, setSelectedDate] = useState<Date>();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<UserEditModel>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      avatarUrl: "",
+      dateOfBirth: undefined,
+      username: "",
+      name: "",
+    },
+    mode: "onSubmit",
+  });
+
+  useEffect(() => {
+    const setDefaultUserData = async () => {
+      if (currentUser) {
+        const userData = await UserController.getUser(currentUser.uid);
+        if (userData) {
+          const matchedAvatar = avatarOptions.find(
+            (option) => option.url === userData!.avatarUrl,
+          );
+          if (matchedAvatar) {
+            setSelectedAvatarOption(matchedAvatar);
+          }
```

Comment:
Nit, do we need the `!` operator? Also, do you think dropping the second if statement and defaulting to the first avatar option if they didn't select one yet works?

```suggestion
          const matchedAvatar = avatarOptions.find(
            (option) => option.url === userData.avatarUrl,
          );
          setSelectedAvatarOption(matchedAvatar || avatarOptions[0]);
```

Reply:
No we don't need it since there is the if statement wrapping it, good catch! Removed and changed to the suggested default logic